extensions/ql-vscode/src/config.ts

```diff
@@ -279,3 +279,8 @@ export const CANARY_FEATURES = new Setting('canary', ROOT_SETTING);
 export function isCanary() {
   return !!CANARY_FEATURES.getValue<boolean>();
 }
+
+/**
+ * Avoids caching in the AST viewer if the user is also a canary user.
+ */
+export const NO_CACHE_AST_VIEWER = new Setting('noCacheAstViewer', ROOT_SETTING);
```

Comment:
Would it make sense to have an `astViewer` or `viewAst` parent setting, and then a `noCache` or `disableCache` underneath it? Not sure if we expect future AST viewer-related options.

Reply:
I like that better. Made the change.